RFS/src/main/java/com/rfs/cms/OpenSearchWorkCoordinator.java

```diff
@@ -0,0 +1,401 @@
+    public static final String COMPLETED_AT_FIELD_NAME = "completedAt";
+    public static final String HEAD_METHOD = "HEAD";
+
+    private final long tolerableClientServerClockDifferenceSeconds;
+    private final AbstractedHttpClient httpClient;
+    private final String workerId;
+    private final ObjectMapper objectMapper;
+    private final Clock clock;
+
+    public OpenSearchWorkCoordinator(AbstractedHttpClient httpClient, long tolerableClientServerClockDifferenceSeconds,
+                                     String workerId) {
+        this(httpClient, tolerableClientServerClockDifferenceSeconds, workerId, Clock.systemUTC());
+    }
+
+    public OpenSearchWorkCoordinator(AbstractedHttpClient httpClient, long tolerableClientServerClockDifferenceSeconds,
+                                     String workerId, Clock clock) {
+        this.tolerableClientServerClockDifferenceSeconds = tolerableClientServerClockDifferenceSeconds;
+        this.httpClient = httpClient;
+        this.workerId = workerId;
+        this.clock = clock;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    @Override
+    public void close() throws Exception {
+        httpClient.close();
+    }
+
+    public void setup() throws IOException {
+        var indexCheckResponse = httpClient.makeJsonRequest(HEAD_METHOD, INDEX_NAME,null, null);
+        if (indexCheckResponse.getStatusCode() == 200) {
+            log.info("Not creating " + INDEX_NAME + " because it already exists");
+            return;
+        }
+        log.atInfo().setMessage("Creating " + INDEX_NAME + " because it's HEAD check returned " +
+                indexCheckResponse.getStatusCode()).log();
+        var body = "{\n" +
+                "  \"settings\": {\n" +
+                "   \"index\": {" +
+                "    \"number_of_shards\": 1,\n" +
+                "    \"number_of_replicas\": 1\n" +
+                "   }\n" +
+                "  },\n" +
+                "  \"mappings\": {\n" +
+                "    \"properties\": {\n" +
+                "      \"" + EXPIRATION_FIELD_NAME + "\": {\n" +
+                "        \"type\": \"long\"\n" +
+                "      },\n" +
+                "      \"" + COMPLETED_AT_FIELD_NAME + "\": {\n" +
+                "        \"type\": \"long\"\n" +
+                "      },\n" +
+                "      \"leaseHolderId\": {\n" +
+                "        \"type\": \"keyword\",\n" +
+                "        \"norms\": false\n" +
+                "      },\n" +
+                "      \"status\": {\n" +
+                "        \"type\": \"keyword\",\n" +
+                "        \"norms\": false\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }\n" +
+                "}\n";
+
+        var response = httpClient.makeJsonRequest(PUT_METHOD, INDEX_NAME, null, body);
+        if ((response.getStatusCode() / 100) != 2) {
```

Comment:
This is cute - but hard to understand on first glance.  If we are going through the trouble of making our own http client abstraction it could at the very least have a helper method for functionality like this

Reply:
E.g. `response.wasNotSuccessful()`